### ISSUE TITLE
Wrong popup width after using Less/More

### ISSUE THREAD

**Issue by Eagle3386** [uBlockOrigin/uBlock-issues#1485]
### Prerequisites
- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
- [ ] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [x] uBlock Origin is the only extension
    - [x] uBlock Origin with default lists/settings
    - [x] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description
Using latest stable uBO, either on Firefox Dev Edition or latest Firefox Nightly, the popup UI's width is wider than needed if previously reduced via clicking `Less`.
Additionally, increasing it back to its full details via clicking `More`, the popup UI's gets screwed up completely.

### A specific URL where the issue occurs
https://github.com/uBlockOrigin/uBlock-issues

### Steps to Reproduce
0. If already modified, click `More` until all details are shown, then dismiss the popup.
1. Open the popup, click `Less` at least 2 times.
2. Dismiss & reopen the popup.
3. Click `More` as often as you clicked `Less` for step 1.

### Expected behavior:
`Less` & `More` are consecutively adapting to less/more occupied space.

### Actual behavior:
After clicking `Less` 3 times:
![image](https://user-images.githubusercontent.com/5111490/106873672-fc7bbe00-66d4-11eb-85be-6f030edf2465.png)

After clicking `More` 3 times:
![image](https://user-images.githubusercontent.com/5111490/106874427-d6a2e900-66d5-11eb-907a-4aa7ffbf93a8.png)

### Your environment
* uBlock Origin version: 1.33.2
* Browser Name and version: Firefox Dev Edition 86.0b5 & Firefox Nightly 87.0a1
* Operating System and version: Windows 10 Pro x64 20H2

**Comment by uBlock-user** [comment]
Do you have any custom themes enabled ?

**Comment by gorhill** [comment]
I can reproduce, the issue is when uBO is pinned in the overflow menu.

The issue is that uBO takes into account viewport width at first render time only, and this works fine. However the resulting behavior when the popup panel is first render may become unsuitable when adding/removing popup panel components.

The workaround is to simply re-open the popup panel once one has settled down on which popup panel components are wanted.

**Comment by Eagle3386** [comment]
> Do you have any custom themes enabled ?

Nope. Tested with all set to defaults - just as stated in the OP.

> I can reproduce, the issue is when uBO is pinned in the overflow menu.

That's indeed the only customization I always do - pinning every extension into the overflow menu.

> The issue is that uBO takes into account viewport width at first render time only, and this works fine. However the resulting behavior when the popup panel is first rendered may become unsuitable when adding/removing popup panel components.

Ah, I see.

> The workaround is to simply re-open the popup panel once one has settled down on which popup panel components are wanted.

Reopening the panel, after `Less` or `More` where clicked, just once more doesn't narrow its width on both machines.
What am I supposed to do to get its width shrunk correctly?

**Comment by gorhill** [comment]
> doesn't narrow its width

Not sure what you mean. What doesn't narrow what width? The viewport of the overflow menu is fixed, and the popup panel of extensions must adjust to this fixed viewport. This is different from when an extension's icon is in the toolbar, in which case the browser will adjust the viewport of the popup panel to whatever the extension requires.

**Comment by Eagle3386** [comment]
> The viewport of the overflow menu is fixed, and the popup panel of extensions must adjust to this fixed viewport. This is different from when an extension's icon is in the toolbar, in which case the browser will adjust the viewport of the popup panel to whatever the extension requires.

That's the piece that I got wrong - for which I'd like to apologize.
Let me rephrase my comment: reopening after adjusting visible components didn't help uBO catching up & stretching its content, i. e. it remains as shown in the OP.

**Comment by gorhill** [comment]
> it remains as shown in the OP.

Well then it's something I am unable to reproduce on my side:

![a](https://user-images.githubusercontent.com/585534/106916542-4c5a8500-66d5-11eb-82f8-0908746b1987.png)

**Comment by Eagle3386** [comment]
> Well then it's something I am unable to reproduce on my side:
> 
> ![a](https://user-images.githubusercontent.com/585534/106916542-4c5a8500-66d5-11eb-82f8-0908746b1987.png)

That's only true for the full details view. If you lessen the amount by 3 levels & then dismiss & reopen the panel, the bug shown in the 1st image of my OP is still there.

**Comment by gwarser** [comment]
Fonts looks tiny on your screenshot. Did you changed some font or zoom settings?

**Comment by gorhill** [comment]
> bug shown in the 1st image of my OP is still there.

It's not a bug, it's by design -- uBO does not know that the viewport is fixed and thus uses only what it needs in such case. The only way to find out if a viewport is fixed is when uBO's popup panel overflow the viewport, which is a condition that can be detected.

**Comment by Eagle3386** [comment]
> Fonts looks tiny on your screenshot. Did you changed some font or zoom settings?

Yes, preferred `Source Sans Pro` as UI font, but doesn't matter as the bug occurs with the default anyway.

> It's not a bug, it's by design -- uBO does not know that the viewport is fixed and thus uses only what it needs in such case. The only way to find out if a viewport is fixed is when uBO's popup panel overflows the viewport, which is a condition that can be detected.

How can misbehavior be `by design`?! Especially since you're stating that it _can_ be detected, hence _can_ be fixed by stretching the UI to remove the awkward empty space?

**Comment by gwarser** [comment]
https://user-images.githubusercontent.com/886325/107194127-f3019700-69ef-11eb-94ac-c0f1f6b10d71.mp4

**Comment by Eagle3386** [comment]
Exactly the behavior I'm experiencing! Thanks a lot, @gwarser! 👏🏻

**Comment by Eagle3386** [comment]
Thanks a lot, @gwarser! 👏🏻 Exactly the issue I'm facing!

**Comment by gorhill** [comment]
Reopening but to be blunt, I won't spend time to fix this: there are already 100s of issues opened, some since long, and any one of them could be argued to be more important than this one for which a workaround exist.

**Comment by gorhill** [comment]
Whoever is interested in fixing the issue here will have to care more than _just_ the overflow menu, the following must not be broken:

- Rendering/behavior of popup panel in Firefox for Android
- Rendering/behavior of popup panel from toolbar icon
- Rendering/behavior of popup panel when opened in standalone tab

**Comment by siweisun0118** [comment]
Hello,
I am a new developer and I would like to work on this issue for practice. Do you have any recommended starting points to look, or maybe a tutorial to learn the code?

**Comment by artemlazaruk** [comment]
Same issue, without doing anything. Firefox 92.0 fresh installation, uBlock in Overflow menu. Checked with 2 machines.
<img width="444" alt="screenshot_2021-09-21_205422" src="https://user-images.githubusercontent.com/47150358/134222994-a3a1be46-ad63-4ad6-a41b-e5bbe4a44b12.png">

**Comment by gwarser** [comment]
Redesigned on Firefox side. uBO panel will pop out from browser interface now https://bugzilla.mozilla.org/show_bug.cgi?id=1783972
